### ISSUE TITLE
Table Scrolling Wrong when updating items with LazyList and ListContainer

### ISSUE THREAD

**Issue by Ramonrune** [vaadin/framework#9125]
Hello,

I'am using a ListContainer and a lazy list to load some data into a MTable.
But when i try to update MTable, scroll bar goes to a wrong area. It only happen's in the middle of table, not in the start or end. Example: update item 41 and scroll bar goes to item 81.

Code:
Add items into table:
List<File> lazy = new LazyList<>(service::fetchTransactions, service::getTransactionCount, service.PAGE_SIZE);
ListContainer<File> listContainer = new ListContainer<File>(File.class, lazy);
table.setContainerDataSource(listContainer);

Update:
table.getItem(itemid).getItemProperty("propertyhere").setValue(status);
table.refreshRowCache();

![problem](https://cloud.githubusercontent.com/assets/16857245/25246296/971b2f6e-25dd-11e7-8531-84a0a5c3e9e7.png)

**Comment by Ramonrune** [comment]
Link to forum with a sample.

https://vaadin.com/forum#!/thread/15708606

**Comment by Ramonrune** [comment]
Anyone?

**Comment by elmot** [comment]
I have just responded in the thread.

**Comment by Ramonrune** [comment]
I did update to vaadin 7.7.9, but still not working :(, but thanks. Suggestions?

**Comment by tsuoanttila** [comment]
Are you using parts of Viritin in your test application? If yes, can you try to replicate this issue using some of the Vaadin Framework containers?

**Comment by Ramonrune** [comment]
Yes, i'am using, but the problem still happens when i try to switch to vaadin framework containers.

MyUI.java
package com.example.lazyloader;

import java.util.List;

import javax.servlet.annotation.WebServlet;

import org.vaadin.viritin.LazyList;
import org.vaadin.viritin.ListContainer;
import org.vaadin.viritin.fields.MTable;

import com.vaadin.annotations.Theme;
import com.vaadin.annotations.VaadinServletConfiguration;
import com.vaadin.data.util.BeanItemContainer;
import com.vaadin.event.ItemClickEvent;
import com.vaadin.event.ItemClickEvent.ItemClickListener;
import com.vaadin.server.VaadinRequest;
import com.vaadin.server.VaadinServlet;
import com.vaadin.ui.Button;
import com.vaadin.ui.Label;
import com.vaadin.ui.Table;
import com.vaadin.ui.TextField;
import com.vaadin.ui.UI;
import com.vaadin.ui.VerticalLayout;

/**
 * This UI is the application entry point. A UI may either represent a browser window 
 * (or tab) or some part of a html page where a Vaadin application is embedded.
 * <p>
 * The UI is initialized using {@link #init(VaadinRequest)}. This method is intended to be 
 * overridden to add component to the user interface and initialize non-component functionality.
 */
@Theme("mytheme")
public class MyUI extends UI {

    @Override
    protected void init(VaadinRequest vaadinRequest) {
        final VerticalLayout layout = new VerticalLayout();
        PeopleService service = new PeopleService();
        
        Table mtable = new Table();
        mtable.addContainerProperty("numero", Integer.class, "");
        mtable.addContainerProperty("nome", String.class, "");
        
        mtable.setSizeFull();
        
       // List<People> lazy = new LazyList<People>(service::fetchTransactions, service::getTransactionCount, service.PAGE_SIZE);
        BeanItemContainer<People> listContainer = new BeanItemContainer(People.class, service.load(0, 1000));
        mtable.setContainerDataSource(listContainer);
        
     
      //  mtable.setContainerDataSource( new BeanItemContainer<>(People.class, service.fetchTransaction()));
        
        mtable.addItemClickListener(new ItemClickListener() {
			
			@Override
			public void itemClick(ItemClickEvent event) {

				mtable.refreshRowCache();
				mtable.select(event.getItemId());
	        	
			}
		});
        
        layout.addComponent(mtable);
      
        setContent(layout);
    }

    @WebServlet(urlPatterns = "/*", name = "MyUIServlet", asyncSupported = true)
    @VaadinServletConfiguration(ui = MyUI.class, productionMode = false)
    public static class MyUIServlet extends VaadinServlet {
    }
}



PeopleService
package com.example.lazyloader;

import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.Currency;
import java.util.Date;
import java.util.LinkedList;
import java.util.List;
import java.util.Locale;

import javax.swing.filechooser.FileView;
import javax.swing.text.MaskFormatter;
import com.vaadin.data.Property.ValueChangeEvent;
import com.vaadin.data.Property.ValueChangeListener;
import com.vaadin.navigator.View;
import com.vaadin.server.FontAwesome;
import com.vaadin.server.VaadinServlet;
import com.vaadin.server.VaadinSession;
import com.vaadin.shared.ui.label.ContentMode;
import com.vaadin.ui.CheckBox;
import com.vaadin.ui.Label;
import com.vaadin.ui.MenuBar.MenuItem;
import com.vaadin.ui.themes.ValoTheme;
import com.vaadin.ui.PopupDateField;

public class PeopleService {

	// private static final FilesService INSTANCE = new FilesService();
	// private final LinkedList<Files> transaction;

	public final Integer PAGE_SIZE = 500;
	private int dataSetSize = 0;

	public PeopleService() {

		dataSetSize = 1000;

		for (int i = 0; i < dataSetSize; i++) {
			list.add(new People(i, "Pessoa " + i));
		}
	}

	public List<People> fetchTransactions(int startIndex) {
		/*
		 * try { Thread.sleep(500); } catch (InterruptedException e) {
		 * e.printStackTrace(); }
		 */

		// System.out.println("1 startIndex: " + Integer.toString(startIndex));

		return fetchTransactions(startIndex, PAGE_SIZE);
	}

	public List<People> fetchTransactions(int startIndex, int num) {

		int endIndex = startIndex + num > dataSetSize ? dataSetSize : startIndex + num;

		return load(startIndex, endIndex);

		// return load(10, 10);

	}

	public int getTransactionCount() {

		return dataSetSize;

	}

	private List<People> list = new ArrayList<>();

	public List<People> load(int start, int limit) {
		System.out.println(start + " --- " + limit);
		List<People> newList = new LinkedList<>();

		while (start < limit) {

			newList.add(list.get(start));
			start++;
		}
		return newList;

	}

}


People
package com.example.lazyloader;

public class People {
	private int number;
	private String name;
	

	public People(int number, String name) {
		super();
		this.number = number;
		this.name = name;
	}
	
	public int getNumber() {
		return number;
	}
	public void setNumber(int number) {
		this.number = number;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	
	
	
	
}

**Comment by Ramonrune** [comment]
Any Suggestions ?
